examples/stable_diffusion_slackbot/README.md

```diff
@@ -0,0 +1,469 @@
+- **Automatically commit human feedback to an Oxen data repository for later use.** We’ll build a lightweight `flask` server to listen to the [Slack Events API,](https://api.slack.com/apis/connections/events-api) and leverage the `oxenai` python package to save and version our data.
+
+Let’s get started!
+
+## Creating a Stable Diffusion Slackbot w/ Modal
+[Modal Labs](https://modal.com/docs/guide/ex/stable_diffusion_slackbot) provides easy, pythonic access to fast and affordable serverless GPU compute. They have an excellent tutorial on how to set up an image generation Slackbot on their infrastructure, so we’ll provide a few guideposts and then turn it over to them: 
+
+1. [GPU selection](https://modal.com/docs/reference/modal.gpu): we used a `T4` Modal instance to run our Slackbot. It was able to generate 3 images in around 7 seconds at 50 inference steps, with a ~2 second cold start. This was perfect for our needs, but Modal has some faster options to explore as well. 
+2. [Model selection](https://modal.com/docs/reference/modal.gpu): we used a [custom model](https://blog.oxen.ai/creating-a-cute-custom-character-with-stable-diffusion-and-dreambooth/) we’d previously fine-tuned with Dreambooth to produce cute cartoon oxen but for a generic Stable Diffusion starting point, we recommend [CompVis/stable-diffusion-v1-4](https://huggingface.co/CompVis/stable-diffusion-v1-4) or [runwayml/stable-diffusion-v1-5](https://huggingface.co/runwayml/stable-diffusion-v1-5), both available on Hugging Face.
+
+With that, here’s their tutorial—happy building! We’ll see you right back here once you’re ready to generate some awesome images.
+
+[Stable Diffusion Slackbot (Modal Labs)](https://modal.com/docs/guide/ex/stable_diffusion_slackbot)
+
+
+## Collecting Human Feedback with Oxen
+
+Now that we’ve got a working image generation Slackbot, it’s time to build the RLHF infrastructure that will allow us to iteratively improve its quality and consistency over time.
+
+**We’ll cover each step in depth below, but if you’d rather jump straight into the code, you can access it [here](https://github.com/Oxen-AI/examples/tree/main/examples/stable_diffusion_slackbot/code).**
+
+### 1. Building our Flask server
+To parse incoming events from our Slackbot, we’ll need an endpoint at which we can accept POST requests. Flask is a great way to get this up and running in no time — check out the docs here.
+```python
+# server.py
+from flask import Flask, request 
+
+app = Flask(__name__)
+
+# POST route to accept Events API notifications from Slack 
+@app.route("/", methods=["POST"])
+def post():
+	print("Receiving the request!")
+	return "Success"
+
+if __name__ == '__main__':
+    app.run(host="0.0.0.0", port=8000, debug=True)
+```
+
+Start up the development server from the command line:
+
+```bash
+python server.py
+```
+
+Using [ngrok](https://ngrok.com/), we can expose this port on a public IP to allow us to receive messages from the Slackbot while developing locally. 
```

Comment:
I actually think we should also use modal here to deploy the serverless flask app, what do you think?

Reply:
I thiiiink we talked about this a while ago but can't totally remember - definitely fine to do so to tighten up the post but I think it's probably not super cost-effective since we'll have to re-download the oxen lib after every slack reaction? and we also don't need GPUs at this stage..but agree that it's awk to tell people to use ngrok and then be like deployment's on you, go check out digital ocean